api/datadoghq/v1alpha2/datadogpodautoscaler_types.go

```diff
@@ -0,0 +1,541 @@
+	// +optional
+	Horizontal *DatadogPodAutoscalerHorizontalStatus `json:"horizontal,omitempty"`
+
+	// CurrentReplicas is the current number of PODs for the targetRef observed by the controller.
+	// +optional
+	CurrentReplicas *int32 `json:"currentReplicas,omitempty"`
+
+	// Conditions describe the current state of the DatadogPodAutoscaler operations.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []DatadogPodAutoscalerCondition `json:"conditions,omitempty"`
+}
+
+// DatadogPodAutoscalerValueSource defines the source of the value used to scale the target resource.
+type DatadogPodAutoscalerValueSource string
+
+const (
+	// DatadogPodAutoscalerAutoscalingValueSource is a recommendation that comes from active autoscaling.
+	DatadogPodAutoscalerAutoscalingValueSource DatadogPodAutoscalerValueSource = "Autoscaling"
+
+	// DatadogPodAutoscalerManualValueSource is a recommendation that comes from manually applying a recommendation.
+	DatadogPodAutoscalerManualValueSource DatadogPodAutoscalerValueSource = "Manual"
+
+	// DatadogPodAutoscalerLocalValueSource is a recommendation that comes from local fallback.
+	DatadogPodAutoscalerLocalValueSource DatadogPodAutoscalerValueSource = "Local"
+)
+
+// DatadogPodAutoscalerHorizontalStatus defines the status of the horizontal scaling
+type DatadogPodAutoscalerHorizontalStatus struct {
+	// Target is the current target of the horizontal scaling
+	Target *DatadogPodAutoscalerHorizontalTargetStatus `json:"target,omitempty"`
+
+	// LastActions are the last successful actions done by the controller
+	LastActions []DatadogPodAutoscalerHorizontalAction `json:"lastActions,omitempty"`
+}
+
+// DatadogPodAutoscalerHorizontalTargetStatus defines the current target of the horizontal scaling
+type DatadogPodAutoscalerHorizontalTargetStatus struct {
+	// Source is the source of the value used to scale the target resource
+	Source DatadogPodAutoscalerValueSource `json:"source"`
+
+	// GeneratedAt is the timestamp at which the recommendation was generated
+	GeneratedAt metav1.Time `json:"generatedAt,omitempty"`
+
+	// Replicas is the desired number of replicas for the resource
+	Replicas int32 `json:"desiredReplicas"`
+}
+
+// DatadogPodAutoscalerHorizontalAction represents an horizontal action done by the controller
+type DatadogPodAutoscalerHorizontalAction struct {
+	// Time is the timestamp of the action
+	Time metav1.Time `json:"time"`
+
+	// FromReplicas is the number of replicas before the action
+	FromReplicas int32 `json:"replicas"`
+
+	// ToReplicas is the effective number of replicas after the action
+	ToReplicas int32 `json:"toReplicas"`
+
+	// RecommendedReplicas is the original number of replicas recommended by Datadog
+	RecommendedReplicas *int32 `json:"recommendedReplicas,omitempty"`
+
+	// LimitedReason is the reason why the action was limited (ToReplicas != RecommendedReplicas)
```

Comment:
```suggestion
	// LimitedReason is the reason why the action was limited (e.g., ToReplicas != RecommendedReplicas)
```

Reply:
The current version is correct.

Sometimes DPA will make some recommendation but apply value different than the recommended one. For example we might recommend 110 replicas but configuration limits the DPA to applying no more than 100. Then this filed will explain the reason. I added a clarification